Fe 5 add searchbar in events view

abdevworks opened the pull request:
Do zrobienia jeszcze obsługa routingu po kliknięciu w rezultat wyszukiwania. Na obecną chwilę nazwy dla routingu nie są jeszcze ustalone.

abdevworks committed:
feat: searchbar

Created layout for searchbar input

abdevworks committed:
Merge branch 'develop' into FE-5-add-searchbar-in-events-view

abdevworks committed:
feat: searchbar

Added basic layout for searchbar dropdown UI

abdevworks committed:
feat: searchbar

Updated icon for text search

abdevworks committed:
feat: searchbar

Changed folder structure for searchbar

abdevworks committed:
feat: searchbar

Added result list to dropdown

abdevworks committed:
feat: searchbar

Added result list to dropdown

abdevworks committed:
feat: searchbar

Updated folder structure for searchbar-ui component. Refactored code to make it more readable

abdevworks committed:
feat: searchbar

Added emitions of search phrase and search type

dominikapap reviewed (approved):
ok, nie znalazłam błędów

akademiaEnvelo commented on src/app/core/theme.component.ts:
```diff
@@ -1,8 +1,61 @@
+  template: `
+    <h1>Storybook-like route</h1>
+    <app-searchbar [searchResults]="searchResults"></app-searchbar>
+    <app-searchbar [searchResults]="searchResults" [typeOfEvent]="false"></app-searchbar>
```
dodajmy nazwe komponentu do wyświetlenia

abdevworks replied:
Nazwa została dodana

akademiaEnvelo commented on src/app/features/ui/searchbar-ui/searchbar-dropdown/searchbar-dropdown.component.scss:
```diff
@@ -0,0 +1,88 @@
+    border-radius: 25px;
+    margin-right: 0.8rem;
+    margin-left: 0.4rem;
+  }
+  img {
```
brakuje przerwy między regułami

abdevworks replied:
Przerwy zostały dodane.

akademiaEnvelo commented on src/app/features/ui/searchbar-ui/searchbar-dropdown/searchbar-dropdown.component.ts:
```diff
@@ -0,0 +1,60 @@
+@Component({
+  selector: "app-searchbar-dropdown",
+  standalone: true,
+  imports: [MatIconModule, CommonModule],
```
zamiast całego commonmodule, zimportujmy te dyrektywy których używamy jak np `NgIf`, `NgFor` itp

abdevworks replied:
Zmiana została wprowadzona

akademiaEnvelo commented on src/app/features/ui/searchbar-ui/searchbar-dropdown/searchbar-dropdown.component.scss:
```diff
@@ -0,0 +1,88 @@
+  border: 1.15px solid $border;
+  border-radius: 15px;
+  padding: 12px;
+}
```
linijka przerwy między regułami zwiększy czytelność styli

abdevworks replied:
Została dodana linia przerwy między regułami.

abdevworks committed:
feat: searchbar

Added header to theme example. Fixed common module imports

rafalfuz commented:
Jak dla mnie to petarda jest to napisane, błędów nie widzę

rafalfuz commented:
> Jak dla mnie to petarda jest to napisane, błędów nie widzę

Przydałby się tylko jeszcze link w opisie
https://fe-5-add-searchbar-in-events-view--melo-envelo.netlify.app/theme

akademiaEnvelo commented on src/app/features/ui/searchbar-ui/searchbar-dropdown/searchbar-dropdown.component.scss:
```diff
@@ -0,0 +1,88 @@
+$result-hover: #473636;
+
+/* searchbar dropdown */
+.searchbarDropdown {
```
`searchbar-dropdown`, w css używamy kebab case

abdevworks replied:
Nazwy klas zostały poprawione.

akademiaEnvelo commented on src/app/features/ui/searchbar-ui/searchbar-dropdown/searchbar-dropdown.component.scss:
```diff
@@ -0,0 +1,88 @@
+$text: #a7a4a5;
+$result-hover: #473636;
+
+/* searchbar dropdown */
```
ten komentarz jest zbędny, klasa nam mówi do czego odnosi się styl poniżej

abdevworks replied:
Komentarz usunięty. Został po tym jak zarówno searchbar jak i dropdown były jeszcze w jednej klasie.

abdevworks committed:
feat: searchbar

Fixed layout for width of 240px. Dropdown shows after 1 sign now

abdevworks committed:
Merge branch 'develop' of https://github.com/akademia-envelo-3/MELO-front into FE-5-add-searchbar-in-events-view

abdevworks committed:
feat: searchbar

Changed how searchbar example is displayed in theme component

abdevworks committed:
feat: searchbar

Made suggested changes from CR

akademiaEnvelo commented on src/app/app.module.ts:
```diff
@@ -21,6 +22,7 @@ import { CustomHttpInterceptor } from './core';
     StoreModule.forRoot({}),
     EffectsModule.forRoot([]),
     BrowserAnimationsModule,
+    MatIconModule,
```
czemu ten moduł ląduje w app.module a nie w module gdzie go rzeczywiście używamy?

abdevworks replied:
Niepotrzebny moduł został usunięty.

akademiaEnvelo commented on src/app/features/ui/searchbar-ui/searchbar-dropdown/searchbar-dropdown.component.scss:
```diff
@@ -0,0 +1,94 @@
+$background: #211e1d;
+$border: #5e4139;
+$icon: #ead63d;
+$text: #a7a4a5;
+$result-hover: #473636;
```
skąd są te wzory kolorów? nie pochodzą z design systemu?

abdevworks replied:
Kolory zostały zmienione na kolory ze styli globalnych.

akademiaEnvelo commented on src/app/features/ui/searchbar-ui/searchbar/searchbar.component.ts:
```diff
@@ -0,0 +1,56 @@
+
+  isEmpty = true;
+  isDropdownNotVisible = true;
+  minNumberOfLettersToShowDropdown = 1;
+  maxNumberOfResultsToDisplay = 6;
```
dodajmy przed nimi `readonly` bo z tego co widzę to są stałe i nie zmieniają się wczasie, dodatkowo jeśli to są stałe możemy przyjąć zapis zmiennej `MIN_NUMBER_OF_LETTERS_TO_SHOW_DROPDOWN`

abdevworks replied:
Stałe zostały zmienione na readonly. Nazwy stałych również zostały zmienione na duże litery zgodnie z sugerowaną konwencją.

akademiaEnvelo commented on src/app/features/ui/searchbar-ui/searchbar/searchbar.component.ts:
```diff
@@ -0,0 +1,56 @@
+    }
+  }
+  private checkToShowSearchbarDropdown() {
+    if (this.searchPhrase.length >= this.minNumberOfLettersToShowDropdown) {
+      this.isDropdownNotVisible = false;
+    } else {
+      this.isDropdownNotVisible = true;
+    }
```
można skrócić do `this.isDropdownNotVisible  = `this.searchPhrase.length < this.minNumberOfLettersToShowDropdown`, ale decyzje pozstawiam wam

abdevworks replied:
Ok. Zmieniłem to zgodnie z sugestią.

abdevworks committed:
feat: searchbar

Removed unused import from app.module. Fixed constants naming and set access to readonly.

abdevworks committed:
Merge branch 'develop' of https://github.com/akademia-envelo-3/MELO-front into FE-5-add-searchbar-in-events-view

abdevworks committed:
Merge branch 'develop' of https://github.com/akademia-envelo-3/MELO-front into FE-5-add-searchbar-in-events-view

abdevworks committed:
feat: searchbar

Changed color styles to be used from global styles

abdevworks committed:
Merge branch 'develop' of https://github.com/akademia-envelo-3/MELO-front into FE-5-add-searchbar-in-events-view

abdevworks committed:
refactor: ui-moports

Fixed shared ui index file and imports in theme. Added folder for spinner dots

abdevworks committed:
feat: searchbar

Changed maximumError value for netlify

akademiaEnvelo commented on angular.json:
```diff
@@ -58,7 +58,7 @@
                   "type": "anyComponentStyle",
                   "maximumWarning": "2kb",
-                  "maximumError": "4kb"
+                  "maximumError": "30kb"
```
w innym cr opisałem ten problem :)

abdevworks replied:
Poprawiliśmy plik styles więc wszystko teraz jest ok.

akademiaEnvelo commented on src/app/features/ui/searchbar-ui/searchbar-dropdown/searchbar-dropdown.component.scss:
```diff
@@ -0,0 +1,96 @@
+@use './src/styles' as st;
```
nie importujmy globalnego arkusza styli do dzieci, użyjmy css variables

abdevworks replied:
Poprawione. Mamy teraz oddzielny, mały plik z kolorami, który importujemy zamiast całego pliku styles.

akademiaEnvelo commented on src/app/features/ui/searchbar-ui/searchbar-dropdown/searchbar-dropdown.component.scss:
```diff
@@ -0,0 +1,96 @@
+    top: 18px;
+    left: 0;
+    font-weight: semi-bold;
+    font-size: 12px;
```
fajnie jakby ten rozmiar pochodził ze zmiennej

abdevworks replied:
Ok. Zmienione tak, aby wielkość czcionki pochodziła z globalnych ustawień typografii.

akademiaEnvelo commented on src/app/features/ui/searchbar-ui/searchbar/searchbar.component.scss:
```diff
@@ -0,0 +1,57 @@
+@use './src/styles' as st;
```
jw

abdevworks replied:
Poprawione jw.

abdevworks committed:
Merge branch 'develop' of https://github.com/akademia-envelo-3/MELO-front into FE-5-add-searchbar-in-events-view

abdevworks committed:
feat: searchbar

Fixed styles import, so now its only colors that are imported

abdevworks committed:
feat: searchbar

Fixed font-size styling for dropdown result

abdevworks committed:
feat: searchbar

Fixed all font-sizes in searchbar so its using values from global typography setting

abdevworks committed:
fix: searchbar

Fixed close button bug and changed placeholder text

abdevworks committed:
Merge branch 'develop' of https://github.com/akademia-envelo-3/MELO-front into FE-5-add-searchbar-in-events-view

abdevworks committed:
Merge branch 'develop' of https://github.com/akademia-envelo-3/MELO-front into FE-5-add-searchbar-in-events-view
